src/github.com/couchbase/sync_gateway/db/crud.go

```diff
@@ -598,7 +602,9 @@ func (db *Database) updateDoc(docid string, allowImport bool, callback func(*doc
 				if curBody, err = db.getAvailableRev(doc, doc.CurrentRev); curBody != nil {
 					base.LogTo("CRUD+", "updateDoc(%q): Rev %q causes %q to become current again",
 						docid, newRevID, doc.CurrentRev)
-					channels, access, roles, err = db.getChannelsAndAccess(doc, curBody, doc.CurrentRev)
+					channelSet, access, roles, oldBodyBytes, err = db.getChannelsAndAccess(doc, curBody, doc.CurrentRev)
+
+					oldBodyJSON = string(oldBodyBytes)
```

Comment:
I think we're already doing a similar []byte to string cast in getChannelsAndAccess.  Can we just return as string directly from getChannelsAndAccess?